url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/FirstPass.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package backend;
+
+import frontend.FrontEnd;
+import java.awt.Color;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.util.Hashtable;
+import java.util.StringTokenizer;
+
+/**
+ *
+ * @author Nikhil
+ */
+public class FirstPass
+{
+
+    private static boolean commentflag = false;
+    public static Hashtable branchtable = new Hashtable();
+    public static Object obj[][];
+    //  static Enumeration check;
+
+    public static int scan(File path)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.